tmail-backend/jmap/extensions-rabbitmq/src/main/java/com/linagora/tmail/james/jmap/RabbitMQEmailAddressContactSubscriber.java

```diff
@@ -0,0 +1,140 @@
+import reactor.rabbitmq.BindingSpecification;
+import reactor.rabbitmq.ExchangeSpecification;
+import reactor.rabbitmq.QueueSpecification;
+import reactor.rabbitmq.Receiver;
+import reactor.rabbitmq.Sender;
+
+public class RabbitMQEmailAddressContactSubscriber implements Closeable {
+    static class RabbitMQConfiguration {
+        public static RabbitMQConfiguration from(Configuration config) {
+            return new RabbitMQConfiguration(
+                config.getString("address.contact.exchange", "AddressContactExchangeDefault"),
+                config.getString("address.contact.queue", "AddressContactQueueDefault"),
+                config.getString("address.contact.routingKey", "AddressContactRoutingKeyDefault"));
+        }
+
+        private final String exchangeName;
+        private final String queueName;
+        private final String routingKey;
+
+        public RabbitMQConfiguration(String exchangeName, String queueName, String routingKey) {
+            Preconditions.checkArgument(StringUtils.isNotBlank(exchangeName));
+            Preconditions.checkArgument(StringUtils.isNotBlank(queueName));
+            Preconditions.checkArgument(StringUtils.isNotBlank(routingKey));
+            this.exchangeName = exchangeName;
+            this.queueName = queueName;
+            this.routingKey = routingKey;
+        }
+
+        public String getExchangeName() {
+            return exchangeName;
+        }
+
+        public String getQueueName() {
+            return queueName;
+        }
+
+        public String getRoutingKey() {
+            return routingKey;
+        }
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RabbitMQEmailAddressContactSubscriber.class);
+
+    private final RabbitMQConfiguration rabbitMQConfiguration;
+    private final ReceiverProvider receiverProvider;
+    private final Sinks.Many<EmailAddressContactMessage> listener;
+    private final EmailAddressContactMessageHandler messageHandler;
+    private final Sender sender;
+    private Disposable listenQueueHandle;
+    private Disposable messageConsume;
+
+    @Inject
+    public RabbitMQEmailAddressContactSubscriber(ReceiverProvider receiverProvider,
+                                                 Sender sender,
+                                                 EmailAddressContactMessageHandler messageHandler,
+                                                 Configuration configuration) {
+        this.rabbitMQConfiguration = RabbitMQConfiguration.from(configuration);
+        this.receiverProvider = receiverProvider;
+        this.messageHandler = messageHandler;
+        this.sender = sender;
+        this.listener = Sinks.many().multicast().directBestEffort();
+    }
+
+    public void start() {
+        sender.declareExchange(ExchangeSpecification.exchange(rabbitMQConfiguration.getExchangeName())).block();
+        sender.declare(QueueSpecification.queue(rabbitMQConfiguration.getQueueName()).durable(DURABLE)).block();
```

Comment:
We should set up dead letter options for this queue.

We might need to create a dedicated queue for dead letter too.